src/Metadata/TypesMatching.php

```diff
@@ -0,0 +1,58 @@
+<?php
+
+declare(strict_types=1);
+
+namespace AutoMapper\Metadata;
+
+use Symfony\Component\PropertyInfo\Type;
+
+/**
+ * @internal
+ *
+ * @extends \SplObjectStorage<Type, Type[]>
```

Comment:
```suggestion
 * @extends \SplObjectStorage<Type, Type>
```
?

Reply:
No it's correct, if you have a field that accept a string or an other type we want to guess the transformer from the first possible type, by example if we have `array` -> `\Traversable|array` as a type we will use `array` as we don't know (yet) how to do a traversable it will be correct since the type allow both

If we want a single relationship we need `[Type, Type][]` this is possible with that but it will change too much current implementation